sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCWriteSuite.scala

```diff
@@ -362,4 +363,80 @@ class JDBCWriteSuite extends SharedSQLContext with BeforeAndAfter {
       assert(sql("select * from people_view").count() == 2)
     }
   }
+
+  test("SPARK-10849: create table using user specified column type.") {
+    val data = Seq[Row](
+      Row(1, "dave", "Boston", "electric cars"),
+      Row(2, "mary", "Seattle", "building planes")
+    )
+    val schema = StructType(
+      StructField("id", IntegerType) ::
+        StructField("first#name", StringType) ::
+        StructField("city", StringType) ::
+        StructField("descr", StringType) ::
+        Nil)
+    val df = spark.createDataFrame(sparkContext.parallelize(data), schema)
+    // Use database specific CHAR/VARCHAR types instead of String data type.
+    val createTableColTypes = "`first#name` VARCHAR(123), city CHAR(20)"
+    assert(JdbcUtils.schemaString(df.schema, url1, Option(createTableColTypes)) ==
+      s""""id" INTEGER , "first#name" VARCHAR(123) , "city" CHAR(20) , "descr" TEXT """)
```

Comment:
This is the case that users specify all the columns. In this case, we should mix the order of the columns. 

In addition, we also need a case that users only specify one/two columns.